File: 232-implement-queue-using-stacks/implement-queue-using-stacks.py
class MyQueue:

    def __init__(self):
        self.first = []
        self.second = []        

    def push(self, x: int) -> None:
        self.first.append(x)       

    def pop(self) -> int:
        while len(self.first) != 0:
            self.second.append(self.first.pop())
        
        popped = self.second.pop()
        while len(self.second) != 0:
            self.first.append(self.second.pop())
        
        return popped      

    def peek(self) -> int:
        while len(self.first) != 0:
            self.second.append(self.first.pop())
        peeked = self.second[-1]
        while len(self.second) != 0:
            self.first.append(self.second.pop())
        
        return peeked
        
    def empty(self) -> bool:
        return len(self.first) == 0
        


# Your MyQueue object will be instantiated and called as such:
# obj = MyQueue()
# obj.push(x)
# param_2 = obj.pop()
# param_3 = obj.peek()
# param_4 = obj.empty()

File: 232-implement-queue-using-stacks/implement-queue-using-stacks.java
class MyQueue {
    Stack<Integer> first;
    Stack<Integer> second;

    public MyQueue() {
        first = new Stack();
        second = new Stack();
    }
    
    public void push(int x) {
        this.first.push(x);
    }
    
    public int pop() {
        while (!this.first.empty()){
            this.second.push(this.first.pop());
        }
        int popped = this.second.pop();

        while (!this.second.empty()){
            this.first.push(this.second.pop());
        }
        return popped;
    }
    
    public int peek() {
        while (!this.first.empty()){
            this.second.push(this.first.pop());
        }
        int peeked = this.second.peek();

        while (!this.second.empty()){
            this.first.push(this.second.pop());
        }
        return peeked;
    }
    
    public boolean empty() {
        return this.first.empty();
    }
}

/**
 * Your MyQueue object will be instantiated and called as such:
 * MyQueue obj = new MyQueue();
 * obj.push(x);
 * int param_2 = obj.pop();
 * int param_3 = obj.peek();
 * boolean param_4 = obj.empty();
 */